core/controllers/question_editor.py

```diff
@@ -68,25 +77,40 @@ class QuestionSkillLinkHandler(base.BaseHandler):
     """A handler for linking and unlinking questions to or from a skill."""
 
     @acl_decorators.can_manage_question_skill_status
-    def post(self, question_id, skill_id):
+    def post(self, question_id, skill_ids):
         """Links a question to a skill."""
-        skill_domain.Skill.require_valid_skill_id(skill_id)
-        skill = skill_services.get_skill_by_id(skill_id, strict=False)
-        if skill is None:
+        try:
+            skill_ids_list = skill_ids.split(',')
+        except Exception:
+            raise self.PageNotFoundException
+
+        if not all(
+                [isinstance(skill_id, basestring)
+                 for skill_id in skill_ids_list]):
+            raise self.PageNotFoundException
+
+        for skill_id in skill_ids_list:
+            skill_domain.Skill.require_valid_skill_id(skill_id)
+
+        try:
+            skill_services.get_multi_skills(skill_ids_list)
+        except:
             raise self.PageNotFoundException(
                 'The skill with the given id doesn\'t exist.')
 
-        # TODO(vinitamurthi): Replace DEFAULT_SKILL_DIFFICULTY
-        # with a value passed from the frontend.
-        question_services.create_new_question_skill_link(
-            question_id, skill_id, constants.DEFAULT_SKILL_DIFFICULTY)
+        for skill_id in skill_ids_list:
+            # TODO(vinitamurthi): Replace DEFAULT_SKILL_DIFFICULTY
+            # with a value passed from the frontend.
+            question_services.create_new_question_skill_link(
+                question_id, skill_id,
+                constants.DEFAULT_SKILL_DIFFICULTY)
         self.render_json(self.values)
 
     @acl_decorators.can_manage_question_skill_status
-    def delete(self, question_id, skill_id):
+    def delete(self, question_id, skill_ids):
         """Unlinks a question from a skill."""
         question_services.delete_question_skill_link(
```

Comment:
Please be careful to make sure that the pluralization filters down correctly -- skill_ids is plural but delete_question_skill_link is singular. Proofread your PR (do a self-review) to check that everything (including the names) makes logical sense -- this will minimize reviewer confusion + additional back-and-forth.

Reply:
Done.